// protoc --go_out=internal/pb --go_opt=paths=source_relative --go-grpc_out=internal/pb --go-grpc_opt=paths=source_relative internal/proto/police.proto
syntax = "proto3";
package proto;

option go_package = "github.com/senpainikolay/pad-sem7/proto"; 


message GetPoliceUserEntry {
    double user_long  = 1; 
    double user_lat = 2; 
    int64 zoom_index =3; 
    string city = 4; 
} 

message PostPoliceEntry {
    double pol_long  = 1; 
    double pol_lat = 2; 
    string city = 3; 
} 

message ConfirmPoliceEntry {
    double pol_long  = 1; 
    double pol_lat = 2; 
    string city = 3; 
    bool confirmation =4; 
} 

message PoliceResponse { 
    bool error =1; 
    string msg = 2; 
}  

message PoliceInfo {  
    double pol_long  = 1; 
    double pol_lat = 2; 
    bool confirmation_notification =3;  
    int64 confirmedBy  =4; 

}

message GetPoliceResponse {  
    repeated PoliceInfo police_info = 1;
}  




message FetchPoliceRequest { GetPoliceUserEntry user_info =1; } 
message PostPoliceRequest { PostPoliceEntry police_info =1; } 
message ConfirmPoliceRequest { ConfirmPoliceEntry police_info =1; }




service PoliceReportingService {
    rpc FetchPolice(FetchPoliceRequest) returns (GetPoliceResponse) {}
    rpc PostPolice(PostPoliceRequest) returns (PoliceResponse) {}
    rpc ConfirmPolice(ConfirmPoliceRequest) returns (PoliceResponse) {} 
  }
   

